zproject/backends.py

```diff
@@ -622,6 +623,39 @@ def email_belongs_to_ldap(realm: Realm, email: str) -> bool:
 ldap_logger = logging.getLogger("zulip.ldap")
 
 
+class LDAPGroupDescriptions(_LDAPUser):
+    """
+    This class is a workaround - we want to use
+    django-auth-ldap to query the ldap directory for
+    group descriptions,but it doesn't provide an API for that or an isolated class for handling
+    the connection. Because connection-handling is tightly integrated
+    into the _LDAPUser class, we have to make this strange inheritance here,
+    in order to be able to comfortably have an ldap connection and make search
+    queries.
+
+    Same reason as LDAPReverseEmailSearch
+    """
+
+    def __init__(self) -> None:
+        # Superclass __init__ requires a username argument - it doesn't actually
+        # impact anything for us in this class, given its very limited use
+        # for only making a search query, so we pass an empty string.
+        super().__init__(LDAPBackend(), username="")
+
+    def get_descriptions(self) -> dict[str, str]:
+        group_description = {}
+        search = settings.AUTH_LDAP_GROUP_SEARCH
+        GROUP_DESCRIPTION_ATTR = settings.LDAP_GROUP_DESCRIPTION_ATTR
+        assert search is not None
+        results = search.execute(self.connection)
+        for group in results:
+            _, ci_dict = group
+            name = ci_dict["cn"][0]
+            description = ci_dict.get(GROUP_DESCRIPTION_ATTR, None)
+            group_description[name] = description
+        return group_description
```

Comment:
So something like this?

```
class ZulipLDAPUser(_LDAPUser):
    """
    This is an extension of the _LDAPUser class, with a realm attribute
    attached to it. It's purpose is to call its inherited method
    populate_user() which will sync the LDAP data with the corresponding
    UserProfile. The realm attribute serves to uniquely identify the UserProfile
    in case the LDAP user is registered to multiple realms.
    """

    def __init__(self, *args: Any, **kwargs: Any) -> None:
        self.realm: Realm = kwargs["realm"]
        del kwargs["realm"]

        super().__init__(*args, **kwargs)

    def _get_group_descriptions(self):
        if self._groups is None:                                                                        
            self._groups = _LDAPUserGroups(self)
        group_infos = self._groups._get_group_infos()
        ldap_group_descriptions ={}
        for group_info in group_infos:
            group_name = self._groups._group_type.group_name_from_info(group_info)
            group_description = self.group_description_from_info(group_info)
            ldap_group_descriptions[group_name] = group_description                                                               
        return ldap_group_descriptions 
    
    def group_description_from_info(self, group_info):
        try:
            description = group_info[1][settings.LDAP_GROUP_DESCRIPTION_ATTR]
        except (KeyError, IndexError):
            description = None
        return description
```

or should the ZulipLDAPUser have additional field for storing the descriptions?

Reply:
Yeah something like that!